File: frontend/src/services/auth/createManagerAccount.ts
import { API_BASE } from '../apiConfig';

export type CreateManagerRequest = {
  name: string;
  email: string;
  password: string;
  phone: string | null;
  avatarUrl: string | null;
  isActive: boolean;
  roles: string[];
};

export type CreateManagerResponse = {
  success: boolean;
};

export async function createManagerAccount(
  payload: CreateManagerRequest,
): Promise<CreateManagerResponse> {
  const res = await fetch(`${API_BASE}/auth/create-manager-account`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(payload),
  });

  if (!res.ok) {
    throw new Error('Failed to create manager account');
  }

  return await res.json();
}

File: backend/gatorsched_api/schemas/auth/create_manager_account.py
from pydantic import BaseModel, EmailStr


class CreateManagerAccountRequest(BaseModel):
    name: str
    email: EmailStr
    password: str
    phone: str | None = None
    avatarUrl: str | None = None
    roles: list[str]


class CreateManagerAccountResponse(BaseModel):
    success: bool
